lib/ClangImporter/ClangImporter.cpp

```diff
@@ -6207,8 +6221,23 @@ TinyPtrVector<ValueDecl *> ClangRecordMemberLookup::evaluate(
           namedMember->getName().getBaseName() != name)
         continue;
 
+      auto memberClangDecl = namedMember->getClangDecl();
+
+      // Skip this base class member if it isn't from recordDecl; this happens
+      // when this member was inherited.
+      if (memberClangDecl != recordDecl->getClangDecl())
+        continue;
+
+      // Skip this base class if this is a case of nested private inheritance.
+      //
+      // BUG: private base class members should be inherited but inaccessible.
+      // Skipping them here may affect accurate overload resolution in cases of
+      // multiple inheritance (which is currently buggy anyway).
```

Comment:
This is a known bug that I wanted to highlight to reviewers. It is a bug in the sense that the behavior differs from what C++ does (`derived.baseMember` complains about `baseMember` being inaccessible/not inherited, rather than being non-existent).

I haven't done so yet but I think the right way to do this is to import those fields with an appropriate `@unavailable` attribute.